### PR TITLE
[WIP]android-ndk: latest

### PR DESCRIPTION
**Pull request by yurikoles**
After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] The commit message includes the cask’s name and version.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference].
- [ ] `brew cask install {{cask_file}}` worked successfully.
- [ ] `brew cask uninstall {{cask_file}}` worked successfully.
- [x] Checked there are no [open pull requests] for the same cask.
- [x] Checked the cask was not already refused in [closed issues].
- [x] Checked the cask is submitted to [the correct repo].

[token reference]: https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md
[open pull requests]: https://github.com/caskroom/homebrew-cask/pulls
[closed issues]: https://github.com/caskroom/homebrew-cask/issues?q=is%3Aissue+is%3Aclosed
[the correct repo]: https://github.com/caskroom/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask

### CODE REVIEW

**Comment by yurikoles** [comment]
I can't create a workaround for cask requiring url stanza.

```sheell
cask install android-ndk
==> Downloading https://developer.android.com/ndk/
######################################################################## 100,0%
==> No checksum defined for Cask android-ndk, skipping verification
Error: Uh oh, could not figure out how to unpack '/Users/yurikoles/Library/Caches/Homebrew/Cask/android-ndk--latest'
Error: nothing to install
```

File: Casks/android-ndk.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+cask 'android-ndk' do
+  version 'latest'
+  sha256 :no_check
+
+  url 'https://developer.android.com/ndk/'
+  name 'Android NDK'
+  homepage 'https://developer.android.com/ndk/'
+
```
Comment (miccal) [inline Casks/android-ndk.rb]: This is missing an [`artifact`](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/all_stanzas.md#at-least-one-artifact-stanza-is-also-required) stanza.

**Review by miccal — CHANGES_REQUESTED** [review]

File: Casks/android-ndk.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+cask 'android-ndk' do
+  version 'latest'
```
Comment (yurikoles) [inline Casks/android-ndk.rb]: done

**Comment by yurikoles** [comment]
There is still an open issue #31234

**Comment by yurikoles** [comment]
@miccal this package is supposed to be meta/fake one that uses `android-sdk`'s `sdkmanager` to install itself.

**Comment by victorpopkov** [comment]
@yurikoles You can try experimenting with the solution below. It will use the shared path provided with `android-sdk.rb` cask to install the `ndk-bundle` as you mentioned earlier:

```ruby
cask 'android-ndk' do
  version '14b'
  sha256 'f5373dcb8ddc1ba8a4ccee864cba2cbdf500b2a32d6497378dfd32b375a8e6fa'

  # google.com/android/repository was verified as official when first introduced to the cask
  url "https://dl.google.com/android/repository/android-ndk-r#{version}-darwin-x86_64.zip"
  name 'android-sdk'
  homepage 'https://developer.android.com/ndk/index.html'

  conflicts_with cask: 'android-platform-tools'
  depends_on cask: 'android-sdk'

  installer script: "#{HOMEBREW_PREFIX}/share/android-sdk/tools/bin/sdkmanager",
            args:   ['ndk-bundle'],
            input:  'y'

  postflight do
    FileUtils.ln_sf("#{HOMEBREW_PREFIX}/share/android-sdk", "#{HOMEBREW_PREFIX}/share/android-ndk")
  end

  uninstall_postflight do
    FileUtils.rm("#{HOMEBREW_PREFIX}/share/android-ndk")
  end

  caveats <<-EOS.undent
    We will install ndk-bundle for you in existing android-sdk directory provided
    by the android-sdk cask.

    You may want to add to your profile:
      'export ANDROID_NDK_HOME=#{HOMEBREW_PREFIX}/share/android-ndk'

    This operation may take up to 10 minutes depending on your internet connection.
    Please, be patient.
  EOS
end
```

Maybe there is a better solution, but at least it should work.

**Comment by yurikoles** [comment]
@victorpopkov the problem with your code is that it makes unwanted download of `android-ndk-r14b-darwin-x86_64.zip` by cask which is later being downloaded by `sdkmanager`.

**Comment by yurikoles** [comment]
@miccal how can we avoid any downloads by cask? I want this cask to not download anything since download will be handled by `sdkmanager`

**Comment by miccal** [comment]
@yurikoles you cannot - Cask by design _must_ download something.

**Comment by yurikoles** [comment]
@miccal could we download some fake data for this cask?

**Comment by miccal** [comment]
Ping @victorpopkov and @vitorgalvao.

**Comment by miccal** [comment]
If it doesn't download anything, then what is the point of adding this to Cask at all?

**Comment by yurikoles** [comment]
@miccal I had tried to download it and make symlinks with no luck in this PR #31614. It looks like we need to write custom wrappers for every binary.

**Pull request closed without merging**